src/nvidia_rag/rag_server/response_generator.py

```
@@ -727,51 +792,144 @@ async def retrieve_summary(
-        # If wait=True, poll for summary with timeout
-        start_time = time.time()
-        while time.time() - start_time < min(3600, timeout):
+        # Non-blocking mode - return NOT_FOUND
+        return {
+            "message": f"Summary for {file_name} not found. To generate a summary, upload the document with generate_summary=true.",
+            "status": "NOT_FOUND",
+            "file_name": file_name,
+            "collection_name": collection_name,
+        }
+
+    except Exception as e:
```

Ruff (0.14.5)

Do not catch blind exception: `Exception`

(BLE001)